src/reader.rs

```diff
@@ -134,8 +145,16 @@ impl NBReader {
                         break;
                     }
                     Ok(_) => {
-                        tx.send(Ok(PipedChar::Char(byte[0])))
-                            .map_err(|_| Error::MpscSendError)?;
+                        if options.strip_ansi_escape_codes && byte[0] == 27 {
+                            while let Ok(_) = reader.read(&mut byte) {
```

Comment:
You're not handing Ok(0) and Err(_) return values correctly here. In case the stream ends in the middle of an escape character.

Would be nice to have a test for that also, if possible :)

Reply:
I've added a test case where the stream ends in the middle of an escape code sequence. The escape character is just a single byte, so that's either there or not.

Is there anything else missing here? I don't think i need special handling of Ok(0) and Err(_) here, because if they occur, the while loop just ends and the outer match clauses apply, i believe.

Reply:
Nice that you added the testcase, thanks :)

I would have liked to provide a test to show my reasoning here, but not sure exactly how to write that.

In the test you provided, after the process is finished, this inner loop is still spinning infinitely, because the inner loop reader.read() returns Ok(0), and the if-condition does not match for the current byte value.

As for the Err-case. I believe the inner loop will terminate, as it's no longer matching the `while let` statement. But I'm not convinced that the next call to reader.read() in the outer loop is guaranteed to return the same Err value?

Hope I'm making myself clear :)

I think the best approach would be to have a bool tracking if we are currently inside an ansi escape code. Though I recognize that might not be so pretty either

Reply:
I had written a longer response, but that seems gone after I accidentally clicked on the failed test results :/ (struggling with your commit msg linter :sweat_smile: ) 

Anyway, I think the Err(_) case *should* be unproblematic, but I am not entirely sure and you are most definitely right about the Ok(0) case. All in all I agree that using a boolean flag here is better than the nested loop. So I just commited that. LMK what you think and thanks for the feedback so far @souze !